flocker/restapi/test/test_infrastructure.py

```diff
@@ -90,6 +211,16 @@ def test_illegalResult(self, logger):
         If the return value of the decorated function cannot be I{JSON} encoded
         then the response generated has the I{INTERNAL SERVER ERROR} code.
         """
+        # Workaround the message validation code in @validateLogging,
+        # otherwise the invalid message being logged throws an exception!
+        # See https://github.com/ClusterHQ/eliot/issues/150
+        def cleanup():
+            for message in logger.messages:
+                if message.get("action_type") == "api:json_request":
+                    if message["json"].get("foo"):
+                        message["json"]["foo"] = None
+        self.addCleanup(cleanup)
```

Comment:
I don't quite get it.  This test produces an illegal JSON response object (it has the un-encodeable `object` instance, specified a couple lines below).  So the server generates an internal error code and logs the encoding error.

The `@validateLogging` decorator verifies the encoding error got logged.

But ... there's still a problem?  Because now the response object also gets logged in the `REQUEST_JSON` success fields.  Or rather, it would - but it can't be encoded there, either.  So... there's a second traceback logged for the problem?  Do we want two tracebacks for the same problem logged every time?